src/oic/oauth2/message.py

```diff
@@ -711,30 +722,30 @@ def verify(self, **kwargs):
         except KeyError:
             _allowed = {}
 
-        for (attribute, (typ, required, _, _, na)) in _spec.items():
+        for (attribute, cparam) in _spec.items():
             if attribute == "*":
                 continue
 
             try:
                 val = self._dict[attribute]
             except KeyError:
-                if required:
+                if cparam.required:
                     raise MissingRequiredAttribute("%s" % attribute)
                 continue
             else:
-                if typ == bool:
+                if cparam.type == bool:
                     pass
                 elif not val:
-                    if required:
+                    if cparam.required:
                         raise MissingRequiredAttribute("%s" % attribute)
                     continue
 
             if attribute not in _allowed:
                 continue
 
-            if isinstance(typ, tuple):
+            if isinstance(cparam.type, tuple):
                 _ityp = None
-                for _typ in typ:
+                for _typ in cparam.type:
                     try:
```

Comment:
I doubt this can raise a ValueError. Not sure though. The entire check is a little weird looking.
Why is this not part of type_check() itself?

If the cparam.type cannot contain a None value, this can be simplified as:

    for _typ in cparam.type:
         try:
             self._type_check(...)
             break
         except ValueError:
             pass
    else:
          raise NotAllowedValue(val)